### PR TITLE
Change route option to be reencrypting.

### PR DESCRIPTION
**Pull request by mwringe**
Changes how the Hawkular Metrics routing works.

In the past we create a passthrough route and the option to specify the Hawkular Metrics certificate directly. This has an issue where the certifiate for the Hawkular Metrics service must contain both the internal hostname and the external one, which can be difficult to obtain since this now requires the user to setup their own intermediate certificate authority. Even using an intermediary CA could also cause problems based on the type of certificates being used.

The docs were updated on how to use a re-encrypting route instead. And this PR allows the re-encrypting route to be automated.

Changes
- the hawkular-metrics.pem secret is now used as the certificate for the Hawkular Metrics re-encrypting route.
- the hawkular-metrics-ca.cert is now used as the 'caCertificate' for the re-encrypting route.

The secrets names here have not been changed, and people who have had this setup in the past should not require any additional changes, it should just continue to work.

- the hawkular-cassandra.pem secret, which could be used to over-ride the self signed certificates for the Cassandra service have been removed. I don't believe anyone was really using this before and I can't think of a good reason to have this exposed as an option.

### CODE REVIEW

**Comment by mwringe** [comment]
@jcantrill @sosiouxme can one of you take a look at this please?

And when did metrics get a travis ci?

File: deployer/scripts/hawkular.sh [hunk 1]
```diff
@@ -187,12 +187,25 @@ EOF
...
-  oc process hawkular-support -v "HAWKULAR_METRICS_HOSTNAME=$hawkular_metrics_hostname" | oc create -f - || true
-  
+  oc process hawkular-support | oc create -f -
+
+  echo "Creating the Hawkular Metrics Route"
+  # We need to create the route after the service has been created so that its labels get applied to the route itself
+  route_params="--hostname=$hawkular_metrics_hostname --service=hawkular-metrics --dest-ca-cert=$dir/hawkular-metrics-ca.cert"
+  if [ -s ${secret_dir}/hawkular-metrics.pem ]; then
+    `openssl rsa -in  ${secret_dir}/hawkular-metrics.pem > $dir/custom-certificate.key`
+    `openssl crl2pkcs7 -nocrl -certfile ${secret_dir}/hawkular-metrics.pem | openssl pkcs7 -print_certs | grep -v "^subject=*\|^issuer=*\|^$" > $dir/custom-certificate.crt`
```
Comment (sosiouxme) [inline deployer/scripts/hawkular.sh]: looks like this is just extracting the cert from the pem, or am i missing something? seems like you could do that more simply:

    openssl x509 -in ${secret_dir}/hawkular-metrics.pem -out $dir/custom-certificate.crt

either way, could probably use a comment as to what this is for

File: hack/keys/create.sh [hunk 2]
```diff
@@ -30,18 +30,21 @@ if [ -n ${OS_ROOT} ]; then
...
 fi
 
+function createCertificate {
+  name=$1
+  hostnames=$2
+  mkdir -p $name
+  oadm ca create-server-cert --cert=$name/$name.cert --key=$name/$name.key --hostnames=$hostnames --signer-cert=${SIGNER_CERT} --signer-key=${SIGNER_KEY} --signer-serial=${SIGNER_SERIAL}
+  cat $name/$name.cert $name/$name.key > $name/$name.pem
+
+  echo "`openssl x509 -in $name/$name.cert; cat $name/$name.key`" > $name/$name-noca.pem
```
Comment (sosiouxme) [inline hack/keys/create.sh]: ( openssl x509 -in $name/$name.cert; cat $name/$name.key ) > $name/$name-noca.pem

File: hack/tests/test_default_deploy.sh [hunk 3]
```diff
@@ -405,65 +405,46 @@ function test.HawkularMetricsCustomCertificate {
   undeployAll
   
   Info "Checking that everything can be properly start with a custom Hawkular Metrics certificate"
```
Comment (sosiouxme) [inline hack/tests/test_default_deploy.sh]: start => started

File: deployer/scripts/hawkular.sh (hunk 1 above)
Comment (mwringe) [inline deployer/scripts/hawkular.sh]: That command does not exactly do what we want it to do. The openssl command assumes a single certificate is in the pem file, which is not always true. The command above will extract all the certificates in the chain as part of the pem file.

When oadm generates certificates, they include both the CA certificate and the server one. This means we can just `cat myServer.crt myServer.key > myServer.pem` and we have all the information we need.

If the user get their certs from something else and their pem only contains the server key and the server certificate itself, they can still add in the certificate chain using the  hawkular-metrics-ca.cert secret.

I will add a comment explaining what we are doing here.

**Comment by sosiouxme** [comment]
nits, otherwise LGTM

**Comment by mwringe** [comment]
Updated the PR to bring in fixes proposed by sosiouxme.

**Pull request merged**